registry/storage/driver/swift/swift.go

```diff
@@ -0,0 +1,519 @@
+	content, err := d.Conn.ObjectGetBytes(d.Container, d.swiftPath(path))
+	if err != nil {
+		return nil, parseError(path, err)
+	}
+	return content, nil
+}
+
+// PutContent stores the []byte content at a location designated by "path".
+func (d *driver) PutContent(ctx context.Context, path string, contents []byte) error {
+	if dir, err := d.createParentFolder(path); err != nil {
+		return parseError(dir, err)
+	}
+	err := d.Conn.ObjectPutBytes(d.Container, d.swiftPath(path),
+		contents, d.getContentType())
+	return parseError(path, err)
+}
+
+// ReadStream retrieves an io.ReadCloser for the content stored at "path" with a
+// given byte offset.
+func (d *driver) ReadStream(ctx context.Context, path string, offset int64) (io.ReadCloser, error) {
+	headers := make(swift.Headers)
+	headers["Range"] = "bytes=" + strconv.FormatInt(offset, 10) + "-"
+
+	file, _, err := d.Conn.ObjectOpen(d.Container, d.swiftPath(path), false, headers)
+
+	if err != nil {
+		if swiftErr, ok := err.(*swift.Error); ok && swiftErr.StatusCode == 416 {
+			return ioutil.NopCloser(bytes.NewReader(nil)), nil
+		}
+
+		return nil, parseError(path, err)
+	}
+
+	return file, nil
+}
+
+// WriteStream stores the contents of the provided io.Reader at a
+// location designated by the given path. The driver will know it has
+// received the full contents when the reader returns io.EOF. The number
+// of successfully READ bytes will be returned, even if an error is
+// returned. May be used to resume writing a stream by providing a nonzero
+// offset. Offsets past the current size will write from the position
+// beyond the end of the file.
+func (d *driver) WriteStream(ctx context.Context, path string, offset int64, reader io.Reader) (int64, error) {
+	var (
+		segments      []swift.Object
+		paddingReader io.Reader
+	)
+
+	partNumber := int64(1)
+	bytesRead := int64(0)
+	currentLength := int64(0)
+	zeroBuf := make([]byte, d.ChunkSize)
+	segmentsContainer := d.Container + "_segments"
+	cursor := int64(0)
+
+	getSegment := func() string {
+		return d.swiftPath(path) + "/" + fmt.Sprintf("%016d", partNumber)
+	}
+
+	max := func(a int64, b int64) int64 {
+		if a > b {
+			return a
+		}
+		return b
+	}
+
+	info, _, err := d.Conn.Object(d.Container, d.swiftPath(path))
+	if err != nil {
+		if swiftErr, ok := err.(*swift.Error); ok {
+			if swiftErr.StatusCode == 404 {
+				// Create a object manifest
+				if dir, err := d.createParentFolder(path); err != nil {
+					return bytesRead, parseError(dir, err)
+				}
+				headers := make(swift.Headers)
+				headers["X-Object-Manifest"] = segmentsContainer + "/" + d.swiftPath(path)
+				manifest, err := d.Conn.ObjectCreate(d.Container, d.swiftPath(path), false, "",
+					d.getContentType(), headers)
+				manifest.Close()
+				if err != nil {
+					return bytesRead, parseError(path, err)
+				}
+			} else {
+				return bytesRead, parseError(path, err)
+			}
+		} else {
+			return bytesRead, parseError(path, err)
+		}
+	} else {
```

Comment:
It seems that `else` can be omitted, because all branches inside the condition end with return.